File: dainf-lab-client/src/app/pages/item/item.ts
import { Identifiable } from '@/shared/crud/crud';
import { Category } from '../category/category';

export interface Item extends Identifiable {
  name: string;
  description: string;
  price: number;
  category: Category;
  assets: Asset[];
  siorg: string;
  location: string;
  quantity: number;
  minimumStock: number;
  type: ItemType;
  images: any
}

export interface Asset extends Identifiable {
  id: number;
  location: string;
  serialNumber: string;
  status: AssetStatus;
}

export type ItemType = 'CONSUMABLE' | 'DURABLE';
export type AssetStatus = 'AVAILABLE' | 'LOANED' | 'RESERVED' | 'UNDER_MAINTENANCE'

File: dainf-lab-server/src/main/java/br/edu/utfpr/dainf/model/Item.java
package br.edu.utfpr.dainf.model;

import br.edu.utfpr.dainf.enums.ItemType;
import br.edu.utfpr.dainf.shared.Identifiable;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.envers.Audited;

import java.math.BigDecimal;
import java.util.List;

@Getter
@Setter
@Entity
@Audited
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "item")
public class Item implements Identifiable<Long> {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @NotBlank
    @NotNull
    @NotEmpty(message = "O campo 'Nome' é obrigatório")
    @Column(name = "name", length = 50, nullable = false)
    private String name;

    @Column(name = "description", columnDefinition = "text")
    private String description;

    @Column(name = "`price`", precision = 19, scale = 2)
    private BigDecimal price;

    @ManyToOne
    @JoinColumn(name = "category_id", referencedColumnName = "id")
    @NotNull(message = "O campo 'Categoria' é obrigatório")
    private Category category;

    @OneToMany(mappedBy = "item", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Asset> assets;

    @OneToMany(mappedBy = "item", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<ItemImage> images;

    private String siorg;

    private String location;

    private BigDecimal minimumStock;

    @Enumerated(EnumType.STRING)
    private ItemType type;
}
